bin/spotxBidAdapter.js

```diff
@@ -292,27 +292,63 @@ function outstreamRender(bid) {
-      script.setAttribute('data-spotx_collapse', '0');
-      script.setAttribute('data-spotx_autoplay', '1');
-      script.setAttribute('data-spotx_blocked_autoplay_override_mode', '1');
-      script.setAttribute('data-spotx_video_slot_can_autoplay', '1');
 
+      const customOverride = utils.getBidIdParameter('custom_override', bid.renderer.config.outstream_options);
+      if (customOverride && utils.isArray(customOverride)) {
+        let contentWidth = false;
+        let contentHeight = false;
+        let contentContainerId = false;
+        utils.logMessage('[SPOTX][renderer] Custom beahavior: please make sure all the options are properly setup.');
+        customOverride.forEach(function(elt) {
+          if (!utils.getBidIdParameter('name', elt) && !utils.getBidIdParameter('value', elt)) {
+            utils.logWarn('[SPOTX][renderer] Custom beahavior: this option is wrong: ' + elt);
+            return;
+          }
+          if (elt.name === 'content_width') {
+            contentWidth = true;
+          }
+          if (elt.name === 'content_height') {
+            contentHeight = true;
+          }
+          if (elt.name === 'content_container_id') {
+            contentContainerId = true;
+          }
+          if (elt.name === 'channel_id' || elt.name === 'vast_url' || elt.name === 'content_page_url' || elt.name === 'ad_unit') {
+            utils.logWarn('[SPOTX][renderer] Custom beahavior: following option cannot be overrided: ' + elt.name);
+            return;
+          }
+          script.setAttribute('data-spotx_' + elt.name, elt.value);
+        });
+        if (!contentWidth) {
+          script.setAttribute('data-spotx_content_width', utils.getBidIdParameter('content_width', bid.renderer.config.outstream_options));
+        }
+        if (!contentHeight) {
+          script.setAttribute('data-spotx_content_height', utils.getBidIdParameter('content_height', bid.renderer.config.outstream_options));
+        }
+        if (!contentContainerId) {
+          script.setAttribute('data-spotx_content_container_id', utils.getBidIdParameter('slot', bid.renderer.config.outstream_options));
+        }
+      } else {
+        utils.logMessage('[SPOTX][renderer] Default beahavior');
+        script.setAttribute('data-spotx_content_width', utils.getBidIdParameter('content_width', bid.renderer.config.outstream_options));
+        script.setAttribute('data-spotx_content_height', utils.getBidIdParameter('content_height', bid.renderer.config.outstream_options));
+        if (utils.getBidIdParameter('ad_mute', bid.renderer.config.outstream_options)) {
+          script.setAttribute('data-spotx_ad_mute', '0');
+        }
+        script.setAttribute('data-spotx_collapse', '0');
+        script.setAttribute('data-spotx_autoplay', '1');
+        script.setAttribute('data-spotx_blocked_autoplay_override_mode', '1');
+        script.setAttribute('data-spotx_video_slot_can_autoplay', '1');
```

Comment:
add all these defaults to if they have customOverride defined. So that, if they only want to override autoplay, they don't also have to override collapse to keep the previous behavior